## XYZ's credit Analysis

Bank lends loans to its customers. The bank manager and all those engaged in lending to clients must have a clear background check on their customers so as to avoid loan defaults. This is an analytic based on credits lent by bank XYZ.  

In [1]:
# import necessary libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [8]:
# Reading table
df=pd.read_csv('credit.csv')

In [11]:
# Select the columns from the table
selected_columns = [
    'Current Loan Amount',
    'Credit Score',
    'Annual Income',
    'Monthly Debt',
    'Months since last delinquent',
    'Number of Credit Problems',
    'Current Credit Balance',
    'Maximum Open Credit',
    'Bankruptcies',
    'Tax Liens'
]

# Create a new DataFrame containing only the selected columns
credit = df[selected_columns]

### EDA

In [13]:
credit.head(3)

Current Loan Amount  Credit Score  Annual Income  Monthly Debt  \
0               611314         747.0      2074116.0      42000.83   
1               266662         734.0      1919190.0      36624.40   
2               153494         709.0       871112.0       8391.73   

   Months since last delinquent  Number of Credit Problems  \
0                           NaN                          0   
1                           NaN                          0   
2                          10.0                          0   

   Current Credit Balance  Maximum Open Credit  Bankruptcies  Tax Liens  
0                  621908              1058970           0.0        0.0  
1                  679573               904442           0.0        0.0  
2                   38532               388036           0.0        0.0

In [20]:
# removing null values from dataset
credit.dropna()

Current Loan Amount  Credit Score  Annual Income  Monthly Debt  \
2                  153494         709.0       871112.0       8391.73   
3                  176242         727.0       780083.0      16771.87   
4                  321992         744.0      1761148.0      39478.77   
9                  266288         683.0      2031518.0      12443.10   
11                 258104         723.0      1284514.0       6368.99   
...                   ...           ...            ...           ...   
9989               110946         736.0       632396.0      13122.16   
9990               260128         611.0      3182519.0      79563.07   
9994               442596         739.0      1528968.0      19494.38   
9996               132550         718.0       763192.0      12401.87   
9998             99999999         721.0       972097.0      12232.20   

      Months since last delinquent  Number of Credit Problems  \
2                             10.0                          0   
3                             27.0                          1   
4                             44.0                          0   
9                             56.0                          2   
11                            64.0                          0   
...                            ...                        ...   
9989                          48.0                          0   
9990                          40.0                          0   
9994                          79.0                          0   
9996                          20.0                          0   
9998                          24.0                          1   

      Current Credit Balance  Maximum Open Credit  Bankruptcies  Tax Liens  
2                      38532               388036           0.0        0.0  
3                     156940               531322           1.0        0.0  
4                     359765               468072           0.0        0.0  
9                      31445               251130           2.0        0.0  
11                    128402               266904           0.0        0.0  
...                      ...                  ...           ...        ...  
9989                  214966               324126           0.0        0.0  
9990                 1398932              3673032           0.0        0.0  
9994                  419235               656876           0.0        0.0  
9996                   74309               329692           0.0        0.0  
9998                  184984               240658           0.0        0.0  

[3738 rows x 10 columns]

In [14]:
credit.shape


(10000, 10)

there are 10,000 rows and 10 columns

In [21]:
credit['Credit Score'].describe()

count    8019.000000
mean     1077.991520
std      1477.467761
min       585.000000
25%       706.000000
50%       725.000000
75%       741.000000
max      7510.000000
Name: Credit Score, dtype: float64

We can deduce that the mean credit score is 725, and the interquartile range (741-585)= 156.
Credit scores typically range from around 300 to 850, although the specific range and scoring model can vary depending on the credit bureau or scoring system used. In most scoring models, higher scores indicate better creditworthiness.

### Do annual income  correlate with the current loan amount?

In [23]:
correlation = credit['Current Loan Amount'].corr(credit['Annual Income'])

In [24]:
correlation

0.009343424108932985

There is a negligible linear relationship between the two variables. This may be because some people tend to solve other financial issues rather than paying debts.

### How many people have credit card problems?

In [26]:
credit['Number of Credit Problems'].describe()

count    10000.000000
mean         0.165500
std          0.500934
min          0.000000
25%          0.000000
50%          0.000000
75%          0.000000
max         10.000000
Name: Number of Credit Problems, dtype: float64

Maximum number of credit problems is 10. while the least is none. The majority have no problems.

With such analysis one can evaluate credit viability of a client in a bank.

## END